Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxComponentMode.h

```diff
@@ -43,32 +35,10 @@ namespace AzToolsFramework
         BoxComponentMode& operator=(const BoxComponentMode&) = delete;
         BoxComponentMode(BoxComponentMode&&) = delete;
         BoxComponentMode& operator=(BoxComponentMode&&) = delete;
-        ~BoxComponentMode();
+        ~BoxComponentMode() = default;
```

Comment:
The destructor of BaseShapeComponent mode is doing stuff that I need. Wouldn't override prevent that from running? Or is there a preferred way of handling that?

Reply:
`override` is purely optional in this case, for virtual destructors it's not a bad idea for clarity, they will always be called

Reply:
Using `override` in the destructor will still call the parent's destructor, that's different from a normal function member polymorphism, but having the `override` has the benefit that it will stop compiling if the base class' destructor is changed to be non-virtual by mistake.